watertap/unit_models/leaky_pressure_exchanger.py

```diff
@@ -0,0 +1,491 @@
+            doc='Mass transfered on high pressure side')
+        def mass_transfer(b, t, p, j):
+            return b.properties_in[t].flow_mass_phase_comp[p,j]-b.properties_out[t].flow_mass_phase_comp[p,j]
+
+        #Build control volume for low pressure side
+        self.low_pressure_side = ControlVolume0DBlock(default={
+            "dynamic": False,
+            "has_holdup": False,
+            "property_package": self.config.property_package,
+            "property_package_args": self.config.property_package_args})
+
+        self.low_pressure_side.add_state_blocks(
+            has_phase_equilibrium=False)
+
+        self.low_pressure_side.add_material_balances(
+            balance_type=self.config.material_balance_type,
+            has_mass_transfer=True)
+
+        self.low_pressure_side.add_momentum_balances(
+            balance_type=self.config.momentum_balance_type,
+            has_pressure_change=True)
+
+        self.low_pressure_side.deltaP.setlb(0)
+
+        @self.low_pressure_side.Expression(
+            self.flowsheet().config.time,
+            doc='Work transferred to low pressure side fluid')
+        def work(b, t):
+            return b.properties_in[t].flow_vol * b.deltaP[t]
+
+        @self.low_pressure_side.Expression(
+            self.flowsheet().config.time,
+            self.config.property_package.phase_list,
+            self.config.property_package.component_list,
+            doc='Mass transfered on low pressure side (will be negative, gained mass)')
+        def mass_transfer(b, t, p, j):
+            return b.properties_in[t].flow_mass_phase_comp[p,j]-b.properties_out[t].flow_mass_phase_comp[p,j]
+
+        #Add Ports
+        self.add_inlet_port(name='high_pressure_inlet', block=self.high_pressure_side)
+        self.add_outlet_port(name='high_pressure_outlet', block=self.high_pressure_side)
+        self.add_inlet_port(name='low_pressure_inlet', block=self.low_pressure_side)
+        self.add_outlet_port(name='low_pressure_outlet', block=self.low_pressure_side)
+
+        # Performance equations
+        @self.Constraint(
+            self.flowsheet().config.time,
+            doc="Pressure transfer")
+        def eq_pressure_transfer(b, t):
+            return (b.low_pressure_side.deltaP[t] ==
+                    b.efficiency_pressure_exchanger[t] * -b.high_pressure_side.deltaP[t])
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            doc="Equal volumetric flow rate")
+        def eq_equal_flow_vol(b, t):
+            return (b.high_pressure_side.properties_in[t].flow_vol ==
+                    b.low_pressure_side.properties_in[t].flow_vol)
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.config.property_package.phase_list,
+            self.config.property_package.component_list,
+            doc="mass leak from HP inlet to LP outlet")
+        def eq_mass_transfer_from_high_to_low_pressure_outlet(b, t, p, j):
```

Comment:
It would be far more efficient to just write:

```
b.high_pressure_side.mass_transfer[p, j] == b.solvent_leakage_fraction[t]*b.high_pressure_side.properties_in[t].flow_mass_phase_comp[p,j]
```